tests/devices/qubit/test_sampling.py

```diff
@@ -420,6 +420,146 @@ def test_var_measure_shot_vector(self, shots, total_copies):
             assert res == 0
 
 
+class TestBroadcasting:
+    """Test that measurements work when the state has a batch dim"""
+
+    def test_sample_measure(self):
+        """Test that broadcasting works for qml.sample and single shots"""
+        rng = np.random.default_rng(123)
+        shots = qml.measurements.Shots(100)
+
+        state = [
+            np.array([[0, 0], [0, 1]]),
+            np.array([[1, 0], [1, 0]]) / np.sqrt(2),
+            np.array([[1, 1], [1, 1]]) / 2,
+        ]
+        state = np.stack(state)
+
+        measurement = qml.sample(wires=[0, 1])
+        res = measure_with_samples(measurement, state, shots, is_state_batched=True, rng=rng)
+
+        assert res.shape == (3, shots.total_shots, 2)
+        assert res.dtype == np.bool8
+
+        # first batch of samples is always |11>
+        assert np.all(res[0] == 1)
+
+        # second batch of samples is either |00> or |10>
+        assert np.all(np.logical_or(res[1] == [0, 0], res[1] == [1, 0]))
+
+        # third batch of samples can be any of |00>, |01>, |10>, or |11>
+        assert np.all(np.logical_or(res[2] == 0, res[2] == 1))
+
+    @pytest.mark.parametrize(
+        "measurement, expected",
+        [
+            (
+                qml.probs(wires=[0, 1]),
+                np.array([[0, 0, 0, 1], [1 / 2, 0, 1 / 2, 0], [1 / 4, 1 / 4, 1 / 4, 1 / 4]]),
+            ),
+            (qml.expval(qml.PauliZ(1)), np.array([-1, 1, 0])),
+            (qml.var(qml.PauliZ(1)), np.array([0, 0, 1])),
+        ],
+    )
+    def test_nonsample_measure(self, measurement, expected):
+        """Test that broadcasting works for the other sample measurements and single shots"""
+        rng = np.random.default_rng(123)
+        shots = qml.measurements.Shots(10000)
+
+        state = [
+            np.array([[0, 0], [0, 1]]),
+            np.array([[1, 0], [1, 0]]) / np.sqrt(2),
+            np.array([[1, 1], [1, 1]]) / 2,
+        ]
+        state = np.stack(state)
+
+        res = measure_with_samples(measurement, state, shots, is_state_batched=True, rng=rng)
+        assert np.allclose(res, expected, atol=0.01)
+
+    @pytest.mark.parametrize(
+        "shots, total_copies",
```

Comment:
i realize you're copying from old tests, but I have since added the `num_copies` property to the `Shots` class, so this isn't needed anymore:)